url1a745d67fc_sobercheg_InterviewTraining_tgz-pJ8-leetcode_oj_Solution2TestJ8/src/careercup/linkedin/KthElementToLast.java

```diff
@@ -0,0 +1,159 @@
+package careercup.linkedin;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Created by Sobercheg on 12/15/13.
+ * <p/>
+ * http://www.careercup.com/question?id=14468873
+ * Write a function that would return the 5th element from the tail (or end)
+ * of a singly linked list of integers, in one pass, and then provide a set
+ * of test cases against that function (please do not use any list manipulation
+ * functions that you do not implement yourself).
+ */
+
+class ListNode<T> {
+    T data;
+    ListNode<T> next;
+
+    ListNode(T data) {
+        this.data = data;
+    }
+
+}
+
+class MyList<T> {
+    ListNode<T> root;
+
+    public MyList<T> add(T data) {
+        if (root == null) {
+            root = new ListNode<T>(data);
+        } else {
+            ListNode<T> newNode = new ListNode<T>(data);
+            newNode.next = root;
+            root = newNode;
+        }
+        return this;
+    }
+
+    public ListNode<T> getRoot() {
+        return root;
+    }
+}
+
+public interface KthElementToLast<T> {
+    public T getKthElementToLast(MyList<T> list, int k);
+}
+
+class KthElementToLastPointers<T> implements KthElementToLast<T> {
+    public T getKthElementToLast(MyList<T> list, int k) {
+        if (list == null || list.getRoot() == null) throw new IllegalArgumentException("List is null");
+        if (k < 0) throw new IllegalArgumentException(String.format("k is negative [%s]", k));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: k (declared at line 50). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 52. Consider formatting this data with &#39;%d&#39; instead.